main/src/cgeo/geocaching/Image.java

```diff
@@ -15,27 +17,92 @@
 
 import java.io.File;
 
+/**
+ * Represent an Image along with Title and Description.
+ * Mostly used for representing image in/from logs.
+ */
 public class Image implements Parcelable {
-    private final String url;
-    private final String title;
-    private final String description;
+    @NonNull private Uri uri;
```

Comment:
@samueltardieu The angular stone I was looking for in the last questions was [`Image.buildUpon()`](https://github.com/cgeo/cgeo/blob/9c0d069de29e876405baf9c43f600ef2369fa23a/main/src/cgeo/geocaching/Image.java#L195-L206)

Found that pattern while studding [`android.net.Uri`](http://developer.android.com/reference/android/net/Uri.html#buildUpon%28%29)